src/tr1d1um/translation/transport_utils.go

```diff
@@ -112,6 +116,47 @@ func decodeValidServiceRequest(services []string, decoder kithttp.DecodeRequestF
 	}
 }
 
+func loadWDMP(encodedWDMP []byte, newCID, oldCID, syncCMC string) (wdmp *setWDMP, err error) {
+	wdmp = new(setWDMP)
+	if err = json.Unmarshal(encodedWDMP, wdmp); err == nil || len(encodedWDMP) == 0 { //len(data) == 0 case is for TEST_SET
+		if err = deduceSET(wdmp, newCID, oldCID, syncCMC); err == nil {
+			if !isValidSetWDMP(wdmp) {
+				err = ErrInvalidSetWDMP
+			}
+		}
+	}
+
+	return
+}
+
+func captureWDMPParameters(ctx context.Context, r *http.Request) (nctx context.Context) {
+	nctx = ctx
+
+	if r.Method == http.MethodPatch {
+		bodyBytes, _ := ioutil.ReadAll(r.Body)
+		r.Body.Close()
+
+		r.Body = ioutil.NopCloser(bytes.NewBuffer(bodyBytes))
+
+		if wdmp, e := loadWDMP(bodyBytes, r.Header.Get(HeaderWPASyncNewCID),
```

Comment:
Alternative, we can load the WDMP and pass the values in the context for both the `decoder` and the `logger` to read - we'd just have to pay the price of coupling them (which goes against the philosophy go-kit recommends)

Reply:
Either way is probably fine.  I would say that if you don't need the parameters in the context itself, just enrich the logger and put the logger into the context.  That would be true, for example, if nothing else but logging needs the parameters.